src/multi_record_log.rs

```diff
@@ -96,8 +96,13 @@ impl MultiRecordLog {
                     MultiPlexedRecord::AppendRecords {
                         queue,
                         records,
-                        position: _,
+                        position,
                     } => {
+                        if !in_mem_queues.contains_queue(queue) {
+                            in_mem_queues
+                                .touch(queue, position, &file_number)
+                                .map_err(|_| ReadRecordError::Corruption)?;
+                        }
```

Comment:
this fixes a bug where after truncating, a reload would complain about inserting in an unknown queue at the very first record of the new first block.